src/main/java/roboy/dialog/states/ordinaryStates/PIAState.java

```diff
@@ -0,0 +1,139 @@
+package roboy.dialog.states.ordinaryStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.context.Context;
+import roboy.context.contextObjects.IntentValue;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.Linguistics;
+import roboy.linguistics.Triple;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.Neo4jRelationships;
+import roboy.memory.nodes.Interlocutor;
+import roboy.dialog.Segue;
+import roboy.util.QAJsonParser;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+import static roboy.memory.Neo4jRelationships.*;
+
+/**
+ * Personal Information Asking State
+ */
+public class PIAState extends State {
+    private QAJsonParser qaValues;
+    private Neo4jRelationships[] predicates = { FROM, HAS_HOBBY, WORK_FOR, STUDY_AT };
+    private Neo4jRelationships selectedPredicate;
+    private State nextState;
+
+    private final String TRANSITION_INFO_OBTAINED = "questionAnswering";
+    final Logger LOGGER = LogManager.getLogger();
+
+    public PIAState(String stateIdentifier, StateParameters params) {
+        super(stateIdentifier, params);
+        qaValues = new QAJsonParser(params.getParameter("qaFile"));
+    }
+
+    @Override
+    public State.Output act() {
+        Interlocutor person = Context.getInstance().ACTIVE_INTERLOCUTOR.getValue();
+
+        for (Neo4jRelationships predicate : predicates) {
+            if (!person.hasRelationship(predicate)) {
+                selectedPredicate = predicate;
+                break;
+            }
+        }
+        List<String> questions = qaValues.getQuestions(selectedPredicate);
+        String question = questions.get((int)(Math.random()*questions.size()));
+        HashMap<String, String> INTENT_VALUE = new HashMap<>();
+        INTENT_VALUE.put("PIA", selectedPredicate.type);
+        Context.getInstance().DIALOG_INTENTS_UPDATER.updateValue(new IntentValue("PIA", selectedPredicate));
```

Comment:
``new IntentValue("PIA"`` 
"PIA" is used as a state identifier here (first argument of ``IntentValue`` is a state identifier). Actually, state identifiers are set in the personality file and can be very different from "PIA". You can get the real identifier by ``getIdentifier()``. 

If you use "PIA" intentionally instead of the real identfier for some reason, please add a comment.

Reply:
Refactored